src/snowflake/connector/constants.py

```diff
@@ -14,6 +14,14 @@
 if TYPE_CHECKING:
     from pyarrow import DataType
 
+from platformdirs import PlatformDirs
+
+dirs = PlatformDirs(
+    appname="snowflake",
+    appauthor=False,
+    ensure_exists=True,
+)
+config_file = dirs.user_config_path / "config.toml"
```

Comment:
Yes, I still need to do this!

Reply:
CC @sfc-gh-jcarroll and @sfc-gh-sdas explicitly for comments on this

Reply:
Wouldn't using `$XDG_CONFIG_DIR` give users more control about where they want their config files, with standardized defaults?

Reply:
In the spec we said the path would be `~/.snowflake/config` and it would be in the ini format (although if it's equivalent to toml in all our docs I think that's fine)

This is the most similar to the current widely used default of `~/.snowsql/config`.

I don't think we should change to a different path now. I think `~/.snowflake/config` is the best option.

I think this will also be 1000x better and more used if it can detect a `~/.snowsql/config` and either accept it or at least give the user some hint to update their path, since every hit when I google "snowflake config file" tells me to put the file at `~/.snowsql/config`.

Reply:
We synced up offline and came up with a new way described [here](https://github.com/snowflakedb/snowflake-connector-python/blob/e88677010ba460f4894f3d6c5ace8a321d406ace/src/snowflake/connector/constants.py#L18)